src/VisualStudio/Core/Def/LanguageService/AbstractPackage.cs

```diff
@@ -5,22 +5,36 @@
 using System;
 using System.Threading;
 using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.ErrorReporting;
 using Microsoft.VisualStudio.ComponentModelHost;
 using Microsoft.VisualStudio.Shell;
 using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.Threading;
 
 namespace Microsoft.VisualStudio.LanguageServices.Implementation.LanguageService;
 
 internal abstract class AbstractPackage : AsyncPackage
 {
-    private IComponentModel? _componentModel_doNotAccessDirectly;
-
     internal IComponentModel ComponentModel
     {
         get
         {
-            Assumes.Present(_componentModel_doNotAccessDirectly);
-            return _componentModel_doNotAccessDirectly;
+            if (field is null)
+            {
+                // We should have been initialized asynchronously, but somebody is asking earlier than we expected, so fetch it synchronously.
+                var componentModel = (IComponentModel?)GetService(typeof(SComponentModel));
+                Assumes.Present(componentModel);
+                field = componentModel;
+                return field;
+            }
+
+            return field;
+        }
```

Comment:
can this just be:

```
get => field ??= (IComponentModel?)GetService(typeof(SComponentModel)) ?? throw etc;
```

Reply:
If we weren't using Assumes.Present, yes....hmmm....